Library won't compile on Uno Wifi Rev 2

- Arduino board:  **Uno Wifi Rev 2**

- Arduino IDE version (found in Arduino -> About Arduino menu):  **1.8.13**

- List the steps to reproduce the problem below (if possible attach a sketch or
  copy the sketch code in too): 
**Open Example sketch "Hello World", select board: uno Wifi Rev 2, Verify Sketch**

In trying to simply compile the Adafruit RGB LCD Shield "Hello World" example, I am getting this error:
```
Arduino: 1.8.12 (Windows Store 1.8.33.0) (Windows 10), Board: "Arduino Uno WiFi Rev2, ATMEGA328"

In file included from C:\Users\...\Arduino\libraries\Adafruit_RGB_LCD_Shield_Library\Adafruit_RGBLCDShield.cpp:17:0:

C:\Users\...\Arduino\libraries\Adafruit_RGB_LCD_Shield_Library\Adafruit_RGBLCDShield.h:68:44: error: expected class-name before '{' token

 class Adafruit_RGBLCDShield : public Print {

                                            ^

exit status 1
Error compiling for board Arduino Uno WiFi Rev2.
```

I think I found some clue to the problem:

Because the uno wifi R2 is a megaavr board and not an avr board, I went looking for the source code for Print.h. These apparently are not in the same location. From what I could find, there are at least two versions. I could not find the avr file (C:\Users\...\ArduinoData\packages\arduino\hardware\avr folder is blank)

However, the megaavr folder was not. I did find Print.h in:
(C:\Users\...\ArduinoData\packages\arduino\hardware\megaavr\1.8.6\cores\arduino\api)

While there, I noticed it defined the "Print" class within the "arduino" namespace.

```
namespace arduino {

class Print
{
```
I thought maybe there's another Print.h in a different namespace that is conflicting.

So, I edited the RGB LCD Shield header file to say:
`class Adafruit_RGBLCDShield : public arduino::Print {`
and it compiles for the Uno Wifi Rev 2. However, the arduino namespace appears to not be known to the avr boards, and now won't compile for them, even if I use 
`using namespace arduino`
in the RGB_LCD header file.

I also checked and this behavior also happens with the liquid crystal library, upon which this is based, so maybe it's their issue to resolve?

--- a/Adafruit_RGBLCDShield.h
+++ b/Adafruit_RGBLCDShield.h
@@ -58,8 +58,8 @@
 #define BUTTON_SELECT 0x01 //!< Select button
 
 #ifdef ARDUINO_ARCH_MEGAAVR 
-using namespace arduino;    //!< MEGA AVR architecture uses the arduino namespace
-#endif                      //!< but AVR arch does not
+using namespace arduino; //!< MEGA AVR architecture uses the arduino namespace
+#endif                   //!< but AVR arch does not
 
 /*!
  * @brief Base class for RGB LCD shield

--- a/Adafruit_RGBLCDShield.h
+++ b/Adafruit_RGBLCDShield.h
@@ -58,8 +58,8 @@
 #define BUTTON_SELECT 0x01 //!< Select button
 
 #ifdef ARDUINO_ARCH_MEGAAVR 
-using namespace arduino;    //!< MEGA AVR architecture uses the arduino namespace in it's Print class definition,
-#endif                      //!< but AVR arch does not, so this pre-compiler code only applies when it needs to.
+using namespace arduino;    //!< MEGA AVR architecture uses the arduino namespace
+#endif                      //!< but AVR arch does not
 
 /*!
  * @brief Base class for RGB LCD shield

--- a/Adafruit_RGBLCDShield.h
+++ b/Adafruit_RGBLCDShield.h
@@ -59,7 +59,7 @@
 
 #ifdef ARDUINO_ARCH_MEGAAVR 
 using namespace arduino;    //!< MEGA AVR architecture uses the arduino namespace in it's Print class definition,
-#endif                      //!<  but AVR arch does not, so this pre-compiler code only applies when it needs to.
+#endif                      //!< but AVR arch does not, so this pre-compiler code only applies when it needs to.
 
 /*!
  * @brief Base class for RGB LCD shield

--- a/Adafruit_RGBLCDShield.h
+++ b/Adafruit_RGBLCDShield.h
@@ -57,6 +57,10 @@
 #define BUTTON_RIGHT 0x02  //!< Right button
 #define BUTTON_SELECT 0x01 //!< Select button
 
+#ifdef ARDUINO_ARCH_MEGAAVR 
+using namespace arduino;    // MEGA AVR architecture uses the arduino namespace in it's Print class definition,
+#endif                      //  but AVR arch does not, so this pre-compiler code only applies when it needs to.
+
 /*!
  * @brief Base class for RGB LCD shield
  */

--- a/Adafruit_RGBLCDShield.h
+++ b/Adafruit_RGBLCDShield.h
@@ -58,8 +58,8 @@
 #define BUTTON_SELECT 0x01 //!< Select button
 
 #ifdef ARDUINO_ARCH_MEGAAVR 
-using namespace arduino;    // MEGA AVR architecture uses the arduino namespace in it's Print class definition,
-#endif                      //  but AVR arch does not, so this pre-compiler code only applies when it needs to.
+using namespace arduino;    //!< MEGA AVR architecture uses the arduino namespace in it's Print class definition,
+#endif                      //!<  but AVR arch does not, so this pre-compiler code only applies when it needs to.
 
 /*!
  * @brief Base class for RGB LCD shield

--- a/Adafruit_RGBLCDShield.h
+++ b/Adafruit_RGBLCDShield.h
@@ -57,7 +57,7 @@
 #define BUTTON_RIGHT 0x02  //!< Right button
 #define BUTTON_SELECT 0x01 //!< Select button
 
-#ifdef ARDUINO_ARCH_MEGAAVR 
+#ifdef ARDUINO_ARCH_MEGAAVR
 using namespace arduino; //!< MEGA AVR architecture uses the arduino namespace
 #endif                   //!< but AVR arch does not
 

